### ISSUE TITLE
bugsnag_meta_data regression

### ISSUE THREAD

**Issue by abraham-chan** [bugsnag/bugsnag-ruby#422]
### Expected behavior
Copy pasting the example from [the docs](https://docs.bugsnag.com/platforms/ruby/rails/customizing-error-reports/#custom-error-diagnostic-data) into a console on v5.4.1 results in the tab appearing correctly on Bugsnag.

### Observed behavior
After updating to v6.6.3 the same code does not create the tab.

### Steps to reproduce
```
class MyCustomError < StandardError
  attr_reader :metadata
  def initialize(message, metadata)
    super(message)
    @metadata = metadata
  end

  def bugsnag_meta_data
     {tabname: metadata}
  end
end

Bugsnag.notify(MyCustomError.new("Error message", value1: '1', value2: {nested: 3}))
```

### Version
6.6.3

**Comment by Cawllec** [comment]
Thanks for the info @abraham-chan, this appears to be a case of the docs lagging behind a little bit.  In version 6 of the notifier there's a metadata class `Bugsnag::MetaData` that should be included into the custom error in order for the middleware to extract metadata.
The example above should actually be:
```
class MyCustomError < StandardError
  include Bugsnag::MetaData
  attr_reader :metadata
  def initialize(message, metadata)
    super(message)
    @metadata = metadata
  end

  def bugsnag_meta_data
     {tabname: metadata}
  end
end

Bugsnag.notify(MyCustomError.new("Error message", value1: '1', value2: {nested: 3}))
```
I'll make sure I get the docs update right away

**Comment by kattrali** [comment]
Hmm, this seems like a mistake though. There's no need to require a module to make this work, as we can just check if the exception responds to `bugsnag_meta_data`.

**Comment by Cawllec** [comment]
In that case I'll create a fix for this to take it back to the v5 behaviour.

**Comment by kattrali** [comment]
Fixed in [v6.6.4](https://github.com/bugsnag/bugsnag-ruby/releases/tag/v6.6.4)
